src/background/api/userInformation.ts

```diff
@@ -0,0 +1,31 @@
+import Axios, { AxiosResponse } from "axios";
+
+import { hostname, userPath } from "./api";
+
+import store from "../redux/store";
+import {updateUser} from "../redux/actions/user";
+import {UserState} from "../redux/actions/userTypes";
+
+export interface UserInformation {
+    userId: number | null;
+    username?: string | null;
+    groups?: number[]|null;
+    password?: string;
+    confirmationPassword?: string;
+}
+
+export const changeUserInformation = (userWithNewInformation: UserInformation): Promise<UserState> => {
+    console.log("User given tu update user api:")
+    console.log(userWithNewInformation)
+    return new Promise((resolve, reject) => {
+        return Axios.put(`${hostname}${userPath}/${userWithNewInformation.userId}/edit`, userWithNewInformation)
+            .then((response: AxiosResponse<UserState>) => {
+                store.dispatch(updateUser(JSON.parse(response.config.data)));
```

Comment:
Why do you parse the json?

Reply:
the changed user is part of the api response in json file format